### DISCUSSION TITLE
Hubspot integration?

### DISCUSSION THREAD

**Discussion by beto-ov** [ComposioHQ/composio#402] [Announcements and Release]
I read in a couple of places that there's a way of integrating Composio with Hubspot but I'm unable to find any action or tool for that. Any clue?

Thanks!

**Comment by eeshsaxena** [comment]
HubSpot integration is available in Composio. Here is how to find and use it:

**Find the HubSpot toolkit:**

```python
from composio_openai import ComposioToolSet, App

toolset = ComposioToolSet()
tools = toolset.get_tools(apps=[App.HUBSPOT])
```

**Connect your HubSpot account:**

```bash
composio add hubspot
```

This will open a browser window to authenticate with your HubSpot account.

**Available HubSpot actions include:**
- Create/update/get contacts
- Create/update deals
- Manage companies
- Log activities and notes
- Search CRM records

**Full example with an agent:**

```python
from composio_openai import ComposioToolSet, App
from openai import OpenAI

client = OpenAI()
toolset = ComposioToolSet()
tools = toolset.get_tools(apps=[App.HUBSPOT])

response = client.chat.completions.create(
    model="gpt-4o",
    tools=tools,
    messages=[{"role": "user", "content": "Create a new contact in HubSpot for John Doe, email john@example.com"}]
)
result = toolset.handle_tool_calls(response)
```

If you do not see HubSpot in the dashboard, make sure you are on the latest version of the composio package:

```bash
pip install composio-core --upgrade
```

You can also browse all available actions at https://app.composio.dev/apps and search for HubSpot.
